tests/test_pdfminer_crypto.py

```diff
@@ -71,3 +72,15 @@ def test_lzwdecode(self):
 class TestRunlength:
     def test_rldecode(self):
         assert rldecode(b"\x05123456\xfa7\x04abcde\x80junk") == b"1234567777777abcde"
+
+
+class TestAES:
+    def test_unpad_aes(self):
+        assert unpad_aes(b"\x10" * 16) == b""
+        assert unpad_aes(b"0123456789abcdef" + b"\x10" * 16) == b"0123456789abcdef"
+        assert unpad_aes(b"0123456789abc\x03\x03\x03") == b"0123456789abc"
+        # NOTE: As per the spec these sorts of things should be padded
+        # with b"\x10" * 16, but it seems reasonable to be robust to the
+        # possibility of false padding bytes as well
```

Comment:
```suggestion
        # NOTE: As per the spec these sorts of things should be padded
        # with b"\x03" * 16, and we test if other types of padding are 
        # not removed.
```

Reply:
No, in the case where the length of the string is a multiple of 16 it should, in theory, *always* be padded with 16 bytes of `\x10`.  I'm guessing that real-world PDF implementations probably don't do this, so the tests here are to verify that if there are trailing bytes of `\x10` or less which aren't actually padding, that they aren't removed.

Reply:
"aren't actually padding" here means that, for example, a final byte of `\x03` isn't preceded by two other `\x03` bytes.

Reply:
Got it!